tests/unit/single_table/test_base.py

```diff
@@ -104,3 +108,584 @@ def test_fit(self):
+            batch_size=3,
+            max_tries=100,
+            conditions=None,
+            transformed_conditions=None,
+            float_rtol=0.01,
+            progress_bar=None,
+            output_file_path=None,
+        )
+
+        # Assert
+        pd.testing.assert_frame_equal(result, sampled_data)
+        rows, conditions, trans_cond, float_rtol, sampled = instance._sample_rows.call_args[0]
+        assert rows == 3
+        assert conditions is None
+        assert trans_cond is None
+        assert float_rtol == 0.01
+        pd.testing.assert_frame_equal(sampled, pd.DataFrame())
+
+    def test__sample_batch_with_sampled_data_bigger_than_batch_size(self):
+        """Test ``sampled_data`` is bigger than the batch size.
+
+        If the sampled data is bigger than the batch size, the returned data frame must be the
+        ``batch_size`` size.
+        """
+        # Setup
+        sampled_data = pd.DataFrame({
+            'name': ['John', 'Doe', 'John Doe', 'John Doe John'],
+            'salary': [80., 60., 100., 300.]
+        })
+        instance = Mock()
+        instance._sample_rows.return_value = (sampled_data, 3)
+
+        # Run
+        result = BaseSynthesizer._sample_batch(
+            instance,
+            batch_size=3,
+            max_tries=100,
+            conditions=None,
+            transformed_conditions=None,
+            float_rtol=0.01,
+            progress_bar=None,
+            output_file_path=None,
+        )
+
+        # Assert
+        pd.testing.assert_frame_equal(result, sampled_data.head(3))
+        rows, conditions, trans_cond, float_rtol, sampled = instance._sample_rows.call_args[0]
+        assert rows == 3
+        assert conditions is None
+        assert trans_cond is None
+        assert float_rtol == 0.01
+        pd.testing.assert_frame_equal(sampled, pd.DataFrame())
+
+    def test__sample_batch_max_tries_reached(self):
+        """Test that when ``max_tries`` is reached, a break occurs."""
+        # Setup
+        sampled_data = pd.DataFrame({
+            'name': ['John', 'Doe', 'John Doe', 'John Doe John'],
+            'salary': [80., 60., 100., 300.]
+        })
+        instance = Mock()
+        instance._sample_rows.return_value = (sampled_data, 3)
+
+        # Run
+        result = BaseSynthesizer._sample_batch(
+            instance,
+            batch_size=10,
+            max_tries=2,
+            conditions=None,
+            transformed_conditions=None,
+            float_rtol=0.01,
+            progress_bar=None,
+            output_file_path=None,
+        )
+
+        # Assert
+        pd.testing.assert_frame_equal(result, sampled_data)
+        rows, conditions, trans_cond, float_rtol, sampled = instance._sample_rows.call_args[0]
+        assert instance._sample_rows.call_count == 2
+
+    def test__sample_batch_storing_output_file(self, tmpdir):
+        """Test that an output file is properly stored while sampling and progress bar updated."""
```

Comment:
minor: can we rephrase to `Test that an output file is properly stored while sampling and that the progress bar is updated`